bdb/cursor.c

```diff
@@ -1601,47 +1601,61 @@ static int bdb_clean_pglog_queue(bdb_state_type *bdb_state,
     struct pglogs_queue_key *qe, *del_qe = NULL;
     struct pglogs_queue_key *curqe = NULL;
 
+#ifdef ASOF_TRACE
+    char *buf;
+    unsigned count = 0;
+#endif
+
     // Grab this in write mode
     // Consumers will grab in read-mode until they anchor against the list by
     // finding an LSN that is greater than their start LSN.
     Pthread_rwlock_wrlock(&queue->queue_lk);
     if (cur)
         curqe = cur->cur;
-    qe = LISTC_BOT(&queue->queue_keys);
+    /* Any orphan relinks at the front can be deleted */
+    qe = LISTC_TOP(&queue->queue_keys);
```

Comment:
You can merge this LISTC_TOP() and one at the end of this loop in while condition itself.